### PR TITLE
Clean up connection reuse code inside Do func as now Go HTTP packge does it internally

### LINKED ISSUE #1646: Clean up TCP connection reuse code inside Do func as now Go HTTP packge does it internally
The changes in the PR #1576 are not needed now as the latest official Go Http package does it internally. So Now, it is possible to remove the code from the PR.

https://github.com/golang/go/blob/master/src/net/http/client.go#L698

### PR DESCRIPTION
**Pull request by itsksaurabh**
The changes in the PR #1576 are not needed now as the latest official Go Http package does it internally. So Now, it is possible to remove the code from the PR.

https://github.com/golang/go/blob/master/src/net/http/client.go#L698

closes #1646

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @itsksaurabh !
LGTM.

Awaiting second LGTM before merging.

**Comment by itsksaurabh** [comment]
Thank you @gmlewis :)

**Review by nightlark — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @nightlark !
Merging.

**Pull request merged**
